From 0ec6b8870fca9b224f1ff7700033ccd59fb04533 Mon Sep 17 00:00:00 2001
From: Jared McArthur <j-mcarthur@ti.com>
Date: Thu, 10 Jul 2025 20:22:43 -0500
Subject: [PATCH 5/5] PENDING: arm64: dts: ti: k3-am62-main: Fix interrupts
 property for csi2rx

Commit 1b959a49f4f5 ("PENDING: arm64: dts: ti: k3-am62-main:
Add interrupts property") added the interrupts property to the
cdns_csi2rx0 node, but added the wrong interrupt IDs. Fix this by
providing the correct interrupt IDs.

Interrupt IDs are taken from the AM62 TRM [0].

Input Index    | Interrupt Source
---------------|--------------------------
gicss0.spi.175 | csi_rx_if.0.csi_err_irq.0
gicss0.spi.173 | csi_rx_if.0.csi_irq.0

[0] https://www.ti.com/lit/pdf/spruiv7

Fixes: 1b959a49f4f5 ("PENDING: arm64: dts: ti: k3-am62-main: Add interrupts property")
Signed-off-by: Jared McArthur <j-mcarthur@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62-main.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62-main.dtsi
index efca30bbea3f..49a566053f4b 100644
--- a/arch/arm64/boot/dts/ti/k3-am62-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62-main.dtsi
@@ -1095,8 +1095,8 @@ ti_csi2rx0: ticsi2rx@30102000 {
 		cdns_csi2rx0: csi-bridge@30101000 {
 			compatible = "ti,j721e-csi2rx", "cdns,csi2rx";
 			reg = <0x00 0x30101000 0x00 0x1000>;
-			interrupts = <GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 156 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "error_irq", "irq";
 			clocks = <&k3_clks 182 0>, <&k3_clks 182 3>, <&k3_clks 182 0>,
 				<&k3_clks 182 0>, <&k3_clks 182 4>, <&k3_clks 182 4>;
-- 
2.50.1

